iceoryx_posh/test/moduletests/test_roudi_iceoyx_roudi_memory_manager.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test.hpp"
+
+#include "iceoryx_posh/roudi/memory/iceoryx_roudi_memory_manager.hpp"
+
+using namespace ::testing;
+
+using iox::roudi::IceOryxRouDiMemoryManager;
+
+
+class IceoryxRoudiMemoryMananger_test : public Test
+{
+  public:
+    IceOryxRouDiMemoryManager* m_roudiMemoryManagerTest{nullptr};
+
+    void SetUp() override
+    {
+    	auto config = iox::RouDiConfig_t().setDefaults();
+    	m_roudiMemoryManagerTest = new IceOryxRouDiMemoryManager(config);
+    }
+
+    void TearDown() override
+    {
+ 	delete m_roudiMemoryManagerTest;
+    }
+};
+
+TEST_F(IceoryxRoudiMemoryMananger_test, IntrospectionMemoryManagerNotInitialzed)
```

Comment:
Almost everything was tested in other tests. I just needed this one to complete.

Reply:
`IceOryxRouDiMemoryManager` is an interface class to `RouDiMemoryManager`, so there are tests missing which ensure that the API calls are redirected to the right class. The coverage somehow indicates that it is covered by these other tests.

Reply:
Yeah, as it is covered. Does it make sense to add more tests ?

Reply:
In then end it make sense to add more tests here, but as i mentioned it is up to you if you want to add them in this PR. 
The problem is that it is hard to make clear which test adds the coverage here and we need to make sure that every class has explicit unit tests.

For the case that you want to do it later, please add an entry in the issue #454 to make visible that this class is not completely tested.

Reply:
For me this PR is done, if more tests are needed, I will add an entry to the issue.

Reply:
@JimmyBch 

> Almost everything was tested in other tests. I just needed this one to complete.

This statement is not correct. You are right, we use this particular class in other tests and if this class would not work properly those test would maybe fail.
But in a unit tests you have the following assumption "Every class works as intended except the class you are testing" and this assumption holds for the other tests where this class is used. And it makes sense to assume that the `IceoryxRoudiMemoryManager` works since we should have a tests for this class where everything is tested and verified.

At the moment this assumption is wrong and with the tests in your PR we should change that.

Reply:
@JimmyBch 
> For me this PR is done, if more tests are needed, I will add an entry to the issue.

This would make sense if you would have touched a lot of files but this is a tiny PR and why not doing it the right way - right away? Quality over quantity - since we are developing safety critical software.